### PR TITLE
[WFLY-18305] Enable dependabot for patch release upgrades.

### PR DESCRIPTION
**Pull request by jamezp**
[WFLY-18305](https://issues.redhat.com/browse/WFLY-18305)

### CODE REVIEW

**Comment by darranl** [comment]
I know you are not touching code but re "No JIRA required." - it seems like a change to how we maintain the project that could be worth a mention in release notes and the Jira gives us that automatic mention.

File: .github/dependabot.yml [hunk 1]
```diff
@@ -2,6 +2,16 @@
...
 version: 2
 updates:
+  - package-ecosystem: "maven"
+    directory: "/"
+    schedule:
+      interval: "daily"
+    ignore:
+       # Only allow for patch release upgrades
+      - dependency-name: '*'
+        update-types: ['version-update:semver-major', 'version-update:semver-minor']
```
Comment (rhusar) [inline .github/dependabot.yml:12]: What's up with mixing single quotes and double quotes in the same yaml file? 😆

File: .github/dependabot.yml [hunk 2]
```diff
@@ -2,6 +2,16 @@
...
+    rebase-strategy: "disabled"
+    open-pull-requests-limit: 5
```
Comment (rhusar) [inline .github/dependabot.yml:14]: I like the conservative 5 here, CI is going to be in demand! :)

File: .github/dependabot.yml (hunk 1 above)
Comment (jamezp) [reply .github/dependabot.yml:12]: That is a copy/paste issue :) We can fix it, but for whatever reason I do it all the time.

**Comment by jamezp** [comment]
> I know you are not touching code but re "No JIRA required." - it seems like a change to how we maintain the project that could be worth a mention in release notes and the Jira gives us that automatic mention.

I'd argue an issue doesn't do much good. All it would really say is "Enable dependabot for dependency upgrades" effectively what the commit message already says. I don't really see a benefit with it, but I can file one.

**Comment by bstansberry** [comment]
> I'd argue an issue doesn't do much good. All it would really say is "Enable dependabot for dependency upgrades" effectively what the commit message already says. I don't really see a benefit with it, but I can file one.

Please do. Thanks. I don't care about the commit message or PR title, but I think this should be recorded.

File: .github/dependabot.yml (hunk 2 above)
Comment (bstansberry) [reply .github/dependabot.yml:14]: Yes. If we can't keep regular merged update activity going with a queue this size we're doing something wrong.

**Review by xstefank — APPROVED** [review]
I would enable it also for minors since they are also expected to be non-breaking.

**Comment by jamezp** [comment]
> I would enable it also for minors since they are also expected to be non-breaking.

I've definitely seen minors that break things. Jackson is a good example of that. I know there are others too so I'm not just picking on them. I'm sure I've done it :)

**Comment by bstansberry** [comment]
Thanks @jamezp, @darranl, @rhusar and @xstefank

**Pull request merged**